### PR TITLE
Serve on custom port

### PR DESCRIPTION
**Pull request by THPubs**
Added the custom port configuration according to #2 !

### CODE REVIEW

**Comment by kasunben** [comment]
Awesome! 😃

File: lib/actions/artisan.js [hunk 1]
```diff
@@ -184,19 +186,63 @@ module.exports = {
...
+    const config = JSON.parse(Utils.readFileSync('./site.json', 'utf-8'));
+    const portToSet = options.port || config.server.port || env.server.defaultPort;
+
+    // If the given port is different than whats in the config, update site.json with the new port
+    if (portToSet !== config.server.port) {
+      config.server.port = portToSet;
+      fsExtra.outputFileSync(`${env.templates.config}`, JSON.stringify(config, null, 2));
+    }
+
+    // Setup the database connection
```
Comment (kasunben) [inline lib/actions/artisan.js]: How about moving this logic into `actions/db.js`?
We can add new method as `db.queryExec()` or like that.

https://github.com/kodeflex/wp-artisan/blob/46dfcb686eceaec1007c2de0eab2a5f76c12ae88/lib/actions/db.js

File: lib/actions/artisan.js [hunk 2]
```diff
@@ -184,19 +186,63 @@ module.exports = {
...
-      message: null,
-      params: {}
+    // Grab the configurations
+    const config = JSON.parse(Utils.readFileSync('./site.json', 'utf-8'));
+    const portToSet = options.port || config.server.port || env.server.defaultPort;
+
+    // If the given port is different than whats in the config, update site.json with the new port
+    if (portToSet !== config.server.port) {
+      config.server.port = portToSet;
+      fsExtra.outputFileSync(`${env.templates.config}`, JSON.stringify(config, null, 2));
```
Comment (kasunben) [inline lib/actions/artisan.js]: Here, we have a method in 'actions/utils.js` as `Utils.writeFileSync()` :)
https://github.com/kodeflex/wp-artisan/blob/46dfcb686eceaec1007c2de0eab2a5f76c12ae88/lib/actions/utils.js#L43

So, we can remove both, `mysql` and `fs-extra` module from here and just call `Utils` and `Db` actions.

Comment (THPubs) [reply lib/actions/artisan.js]: I did tried to use `writeFileSync` but it expects the configs in a different format. Right now, the config object have two inner objects called `db` and `server` but `writeFileSync` expects an object without these two (Their values in the main object).

For example, here's the config I can pass :

```
{
  "siteName": "a",
  "siteVersion": "1.0.0",
  "cliVersion": "0.0.1",
  "db": {
    "mysqlDbHost": "127.0.0.1",
    "mysqlDbPort": "3306",
    "mysqlDbUser": "root",
    "mysqlDbPwd": "earthsoftp",
    "mysqlDbName": "abc2",
    "mysqlDbTblPrefix": "wp_"
  },
  "server": {
    "host": "localhost",
    "port": "8081"
  }
}
```

But here's what `writeFileSync` expects :

```
{ siteName: 'a',
  mysqlDbHost: '127.0.0.1',
  mysqlDbPort: '3306',
  mysqlDbUser: 'root',
  mysqlDbPwd: 'earthsoftp',
  mysqlDbName: 'abc',
  mysqlDbTblPrefix: 'wp_',
  serverHost: 'localhost',
  serverPort: '8081',
  doInstall: false,
  cliVersion: '0.0.1',
  siteUrl: 'http://localhost:8081' }
```

Might have to convert the config before passing :smile: Let's see what I can do...

Comment (kasunben) [reply lib/actions/artisan.js]: ah! got it. :)

Let's add a new method as `writeJSONSync()` or what ever like that in the `Utils`. What you think? :)

Comment (THPubs) [reply lib/actions/artisan.js]: Cool that will work! I will do it now...

**Comment by THPubs** [comment]
Just updated with the changes :smile:

**Comment by THPubs** [comment]
Wait need to update the readme

**Comment by THPubs** [comment]
Ah looks like the whole readme need to be updated right? Then it's for later i think 😄

**Comment by kasunben** [comment]
> Ah looks like the whole readme need to be updated right? Then it's for later i think

haah 😄, yep. let's do it later.

**Pull request merged**